test-suite/squidconf/time_units

```diff
@@ -14,35 +14,35 @@
 dns_timeout 1 millisecond
 dns_timeout 0.001 second
 
-# maximum checks
-dns_timeout 9223372036854 milliseconds
-dns_timeout 9223372036 seconds
-dns_timeout 153722867 minutes
-dns_timeout 2562047 hours
-dns_timeout 106751 days
-dns_timeout 15250 weeks
-dns_timeout 7625 fortnights
-dns_timeout 3558 months
-dns_timeout 292.27 years
-dns_timeout 29.227 decades
-
-
+# check support for the maximum value (expressed in each supported time unit)
+# XXX: Use a 32-bit maximum to avoid breaking tests on 32-bit platforms.
+# TODO: Generate these test lines based on the target build platform.
+dns_timeout 2147483647854 milliseconds
+dns_timeout 2147483647 seconds # 2^31-1
+dns_timeout 35791394 minutes # floor(dns_timeout seconds/60)
+dns_timeout 596523 hours # floor(dns_timeout minutes/60)
+dns_timeout 24855 days # floor(dns_timeout hours/24)
+dns_timeout 3550 weeks # floor(dns_timeout days/7)
+dns_timeout 1775 fortnights # floor(dns_timeout weeks/2)
+dns_timeout 828.5 months # dns_timeout days/30
+dns_timeout 68.095 years # approx(dns_timeout days/365)
+dns_timeout 6.8095 decades
 # a second-precision parameter (time-units)
 
 # minimum checks
 max_stale 1 second
 max_stale 0.0167 minute
 
-# maximum checks
-max_stale 9223372036 seconds
-max_stale 153722867 minutes
-max_stale 2562047 hours
-max_stale 106751 days
-max_stale 15250 weeks
-max_stale 7625 fortnights
-max_stale 3558 months
-max_stale 292.27 years
-max_stale 29.227 decades
+# maximum checks (see similar dns_timeout checks for details)
+max_stale 2147483647 seconds # 2^31-1
+max_stale 35791394 minutes # floor(max_stale seconds/60)
+max_stale 596523 hours # floor(max_stale minutes/60)
+max_stale 24855 days # floor(max_stale hours/24)
+max_stale 3550 weeks # floor(max_stale days/7)
+max_stale 1775 fortnights # floor(max_stale weeks/2)
+max_stale 828.5 months # max_stale days/30
+max_stale 68 years # approx(max_stale days/365)
+max_stale 6.8 decades
```

Comment:
Is there a reason why the two last `max_stale` values differ (slightly) from the two last `dns_timeout` values?